### PR TITLE
Postgres tsvector Support

### PR DESCRIPTION
**Pull request by s0kil**
Closes https://github.com/digitallyinduced/ihp/issues/286
Supersedes https://github.com/digitallyinduced/ihp/pull/344

### CODE REVIEW

File: IHP/Postgres/TSVector.hs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Copyright: (c) digitally induced GmbH, 2021
+-}
+module IHP.Postgres.TSVector where
+
+import BasicPrelude
+
+-- | Represents a Postgres tsvector
+--
+-- See https://www.postgresql.org/docs/current/datatype-textsearch.html
+type TSVector = Text
```
Comment (mpscholten) [inline IHP/Postgres/TSVector.hs:13]: This needs to be a newtype

Comment (mpscholten) [reply IHP/Postgres/TSVector.hs:13]: I don't think a string representation is the right choice here.

This is a valid tsvector: `a:1A fat:2B,4C cat:5D`

The `:` suffix has special meaning here. A good structure would be:

```haskell
data TSVector = TSVector [Lexeme]
data Lexeme = Lexeme { token :: Text, position :: Int, weight :: Text }
```

**Comment by s0kil** [comment]
@mpscholten It is good to go.

File: IHP/Postgres/TSVector.hs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+                position <- double
+                -- The Default Weight Is `D` So Postgres Does Not Include It In The Result
+                weight <- option 'D' $ choice [char 'A', char 'B', char 'C', char 'D']
+                pure $ LexemeRanking { position = truncate position, weight }
+
+            pure $ Lexeme { token = cs token, ranking }
+
+instance ToField TSVector where
+    toField = undefined
```
Comment (mpscholten) [inline IHP/Postgres/TSVector.hs:65]: When we have a FromField implementation maybe we could also have a ToFiedl?

**Review by mpscholten — COMMENTED** [review]
Could you also add some tests?

**Comment by s0kil** [comment]
I've added the tests.

**Review by mpscholten — APPROVED** [review]

**Pull request merged**